src/Components/CountingFruit.js

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+const CountingFruit = ({fruitImage, fruitImageText, fruitName, fruitCounter, setFruitCounter}) => {
+
+    return (
+        <>
+            <div>
+                <img src={fruitImage} alt={fruitImageText}/>
+                <h2>{fruitName}</h2>
+                <button onClick={() => setFruitCounter(fruitCounter - 1)}
+                        disabled={fruitCounter === 0}
+                >
+                    -
+                </button>
+                {fruitCounter}
+                <button onClick={() => setFruitCounter(fruitCounter + 1)}>+</button>
```

Comment:
Nu zal telkens wanneer `fruitCounter` veranderd een onnodige rerender van deze functie plaatsvinden. Wat je zou kunnen doen is de huidige waarde gebruiken die je vanuit de setFruitCounter functie al krijgt:
```suggestion
                <button onClick={() => setFruitCounter(currentValue => currentValue + 1)}>+</button>
```